sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/orc/OrcPartitionReaderFactory.scala

```diff
@@ -98,8 +98,7 @@ case class OrcPartitionReaderFactory(
       new EmptyPartitionReader[InternalRow]
     } else {
       val (requestedColIds, canPruneCols) = resultedColPruneInfo.get
-      val resultSchemaString = OrcUtils.orcResultSchemaString(canPruneCols,
-        dataSchema, resultSchema, partitionSchema, conf)
+      OrcUtils.orcResultSchemaString(canPruneCols, dataSchema, resultSchema, partitionSchema, conf)
```

Comment:
https://github.com/apache/spark/blob/4a3200b08ac3e7733b5a3dc7271d35e6872c5967/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/orc/OrcUtils.scala#L240-L253

@HyukjinKwon The previous failed test cases are mainly due to  `OrcUtils.orcResultSchemaString` will set `resultSchemaString`  to `conf`, so only ignore useless return value `val resultSchemaString` here .

Reply:
👌